app/kuma-dp/cmd/run_test.go

```diff
@@ -94,12 +96,23 @@ var _ = Describe("run", func() {
 
 			// given
 			rootCtx := DefaultRootContext()
-			rootCtx.BootstrapGenerator = func(_ string, cfg kumadp.Config, _ envoy.BootstrapParams) ([]byte, error) {
+			rootCtx.BootstrapGenerator = func(_ string, cfg kumadp.Config, _ envoy.BootstrapParams) (*envoy_bootstrap_v3.Bootstrap, []byte, error) {
```

Comment:
API v4 doesn't seem like happen in the near future https://github.com/envoyproxy/envoy/issues/16456#issuecomment-882187236, but I agree it's better to avoid having such dependencies. I can create an issue since it will be breaking change and we have to return different bootstrapResponses depending on kuma-dp version, so it's better to implement such thing in another PR

Reply:
We can return different responses on different `accept` header :)

Reply:
https://github.com/kumahq/kuma/issues/3756